modules/javafx.controls/src/test/java/test/javafx/scene/control/skin/VirtualFlowTest.java

```diff
@@ -1207,17 +1226,177 @@ public void testSheetChildrenRemainsConstant() {
         assertEquals("Wrong number of sheet children after removing all items", 12, sheetChildrenSize);
     }
 
+    private ArrayLinkedListShim<GraphicalCellStub> circlelist = new ArrayLinkedListShim<GraphicalCellStub>();
+
+    private VirtualFlowShim createCircleFlow() {
+        // The second VirtualFlow we are going to test, with 7 cells. Each cell
+        // contains a Circle whith a radius that varies between cells.
+        VirtualFlowShim<IndexedCell> circleFlow;
+        circleFlow = new VirtualFlowShim();
+
+        circleFlow.setVertical(true);
+        circleFlow.setCellFactory(p -> new GraphicalCellStub() {
+            @Override
+            protected double computeMinWidth(double height) {
+                return computePrefWidth(height);
+            }
+
+            @Override
+            protected double computeMaxWidth(double height) {
+                return computePrefWidth(height);
+            }
+
+            @Override
+            protected double computePrefWidth(double height) {
+                return super.computePrefWidth(height);
+            }
+
+            @Override
+            protected double computeMinHeight(double width) {
+                return computePrefHeight(width);
+            }
+
+            @Override
+            protected double computeMaxHeight(double width) {
+                return computePrefHeight(width);
+            }
+
+        });
+        circleFlow.setCellCount(7);
+        circleFlow.resize(300, 300);
+        circleFlow.layout();
+        circleFlow.layout();
+        return circleFlow;
+    }
+
+    // when moving the flow in one direction, the position of the flow
+    // should not increase in the opposite direction
+    @Test
+    public void testReverseOrder() {
+        double orig = flow.getPosition();
+        flow.scrollPixels(10);
+        double pos = flow.getPosition();
+        assertFalse("Moving in positive direction should not decrease position", pos < orig);
+        flow.scrollPixels(-50);
+        double neg = flow.getPosition();
+        assertFalse("Moving in negative direction should not decrease position", neg > pos);
+    }
+
+    @Test
+    public void testReverseOrderForCircleFlow() {
+        VirtualFlowShim vf = createCircleFlow();
+        double orig = vf.getPosition();
+        vf.scrollPixels(10);
+        double pos = vf.getPosition();
+        assertFalse("Moving in positive direction should not decrease position", pos < orig);
+        vf.scrollPixels(-50);
+        double neg = vf.getPosition();
+        assertFalse("Moving in negative direction should not decrease position", neg > pos);
```

Comment:
"decrease" in log message should be "increase"

Reply:
typos fixed